core/domain/exp_jobs_one_off.py

```diff
@@ -490,3 +493,109 @@ def reduce(key, values):
+                language_codes_to_audio_metadata = voiceovers_mapping.values()
+                for language_codes in language_codes_to_audio_metadata:
+                    for audio_metadata in language_codes.values():
+                        # Get files using the filename.
+                        filename = audio_metadata['filename']
+                        if audio_metadata['duration_secs'] == 0.0:
+                            try:
+                                fs = (fs_domain.AbstractFileSystem(
+                                    fs_domain.GcsFileSystem(
+                                        AUDIO_ENTITY_TYPE,
+                                        item.id)))
+                                raw = (
+                                    fs.get('%s/%s' % (AUDIO_FILE_PREFIX,
+                                                      filename)))
+                                # Get the audio-duration from file use Mutagen.
+                                tempbuffer = python_utils.string_io()
+                                tempbuffer.write(raw)
+                                tempbuffer.seek(0)
+                                # Loads audio metadata with Mutagen.
+                                audio = mp3.MP3(tempbuffer)
+                                tempbuffer.close()
+                                # Fetch the audio file duration from the Mutagen
+                                # metadata.
+                                audio_metadata['duration_secs'] = (
+                                    audio.info.length)
+                                change_count += 1
+                            except Exception as e:
+                                logging.error(
+                                    'MP3 audio file exception for file %s ,'
+                                    'EXP_ID: %s,'
+                                    'STATE_NAME: %s,'
+                                    'REASON: %s' %
+                                    (filename, item.id, state, e))
+                                failed_count += 1
+                        else:
+                            unchanged_count += 1
+                if change_count > 0:
+                    # Create commits to update the exploration.
+                    commit_cmds.append(exp_domain.ExplorationChange({
+                        'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                        'property_name': (
+                            exp_domain.STATE_PROPERTY_RECORDED_VOICEOVERS),
+                        'state_name': state,
+                        'new_value': {
+                            'voiceovers_mapping': voiceovers_mapping
+                        }
+                    }))
+            if change_count > 0:
+                exp_services.update_exploration(
+                    feconf.MIGRATION_BOT_USERNAME, item.id, commit_cmds,
+                    'Update duration_secs for each voiceover recording '
```

Comment:
"Update duration in seconds for each ..."

(Write this human-readably, since people can read it.)

Reply:
Ok will change this.

Reply:
done